tests/test_indices.py

```diff
@@ -251,38 +249,74 @@ def test_simple(self):
 
 class TestTxMaxTxMinIndices:
 
-    def tmax_tmin_time_series(self, values):
-        coords = pd.date_range('1/1/2000', periods=len(values), freq=pd.DateOffset(days=1))
-        tas_plus10 = xr.DataArray(values + 10, coords=[coords, ], dims='time',
-                                  attrs={'standard_name': 'air_temperature',
-                                         'cell_methods': 'time: maximum within days',
-                                         'units': 'K'})
-        tas_minus10 = xr.DataArray(values - 10, coords=[coords, ], dims='time',
-                                   attrs={'standard_name': 'air_temperature',
-                                          'cell_methods': 'time: minimum within days',
-                                          'units': 'K'})
-        return tas_plus10, tas_minus10
-
-    def test_static_daily_temperature_range(self):
-        temp_values = np.random.uniform(-30, 30, size=1095)
-        tasmax, tasmin = self.tmax_tmin_time_series(temp_values + K2C)
+    def random_tmax_tmin_setup(self, length, tasmax_series, tasmin_series):
+        max_values = np.random.uniform(-20, 40, length)
+        min_values = []
+        for i in range(length):
+            min_values.append(np.random.uniform(-40, max_values[i]))
+        tasmax = tasmax_series(np.add(max_values, K2C))
+        tasmin = tasmin_series(np.add(min_values, K2C))
+        return tasmax, tasmin
+
+    def static_tmax_tmin_setup(self, tasmax_series, tasmin_series):
+        max_values = np.add([22, 10, 35.2, 25.1, 18.9, 12, 16], K2C)
+        min_values = np.add([17, 3.5, 22.7, 16, 12.4, 7, 12], K2C)
+        tasmax = tasmax_series(max_values)
+        tasmin = tasmin_series(min_values)
+        return tasmax, tasmin
+
+    def test_random_daily_temperature_range(self, tasmax_series, tasmin_series):
+        days = 365
+        tasmax, tasmin = self.random_tmax_tmin_setup(days, tasmax_series, tasmin_series)
+        dtr = xci.daily_temperature_range(tasmax, tasmin, freq="YS")
+
+        np.testing.assert_array_less([0], [dtr.mean()])
```

Comment:
assert_array_almost_equal(x, y, decimal=6, err_msg='', verbose=True)
x : array_like
    The actual object to check.
y : array_like
    The desired, expected object.